### PR TITLE
[Feature] Runtime Mappings

### PR DESCRIPTION
**Pull request by stefnats**

### CODE REVIEW

**Comment by stefnats** [comment]
When this is merged, `elastic-scout-driver-plus` can be merged to support this as well. I admit, it is difficult to develop for multiple depending repositories.

File: tests/Unit/Search/SearchRequestTest.php [hunk 1]
```diff
@@ -306,6 +306,45 @@ public function test_array_casting_with_post_filter(): void
         ], $request->toArray());
     }
 
+    public function test_array_casting_with_runtime_mapping(): void
```
Comment (babenkoivan) [inline tests/Unit/Search/SearchRequestTest.php:309]:
```suggestion
    public function test_array_casting_with_runtime_mappings(): void
```

File: tests/Unit/Search/SearchRequestTest.php [hunk 2]
```diff
@@ -306,6 +306,45 @@ public function test_array_casting_with_post_filter(): void
...
+    {
+        $request = new SearchRequest([
```
Comment (babenkoivan) [inline tests/Unit/Search/SearchRequestTest.php:311]:
```suggestion
        $request = new SearchRequest();
```

File: tests/Unit/Search/SearchRequestTest.php [hunk 3]
```diff
@@ -306,6 +306,45 @@ public function test_array_casting_with_post_filter(): void
...
+                    'type' => 'keyword',
+                    'script' => 'for(i=0;i<5;i++) echo "hi"'
+                ],
+                'field2' => [
+                    'type' => 'keyword',
+                    'script' => 'for(i=0;i<10;i++) echo "hi"'
+                ],
+            ]
+        ]);
+        $this->assertEquals([
```
Comment (babenkoivan) [inline tests/Unit/Search/SearchRequestTest.php:327]:
```suggestion

        $this->assertEquals([
```

**Review by babenkoivan — CHANGES_REQUESTED** [review]

**Comment by babenkoivan** [comment]
Hey @stefnats, thanks for the PR! I've made some suggestions (all are small things related to the code style). 
Could you please also resolve merge conflicts and add usage examples in README?

**Pull request closed without merging**